thriftbp/prometheus.go

```diff
@@ -0,0 +1,38 @@
+package thriftbp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	thriftLabels = []string{
+		"thrift_service",
+		"thrift_method",
+		"thrift_success",
+		"thrift_exception_type",
+		"thrift_baseplate_status",
+		"thrift_baseplate_status_code",
+	}
+
+	latencyDistribution = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "thrift_server_handling_seconds",
+		Help:    "RPC latencies",
+		Buckets: prometheus.ExponentialBuckets(0.0001, 1.5, 26), // 100us ~ 2.5s
+	}, thriftLabels)
+
+	rpcStatusCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "thrift_server_handled_total",
+		Help: "Total RPC request count",
+	}, thriftLabels)
+
```

Comment:
🔕 It might be helpful to separate these into two var blocks as a way of scoping the labels and the metrics that share them:

```suggestion
)

var (
```